### ISSUE TITLE
support for ip association in classic model

### ISSUE THREAD

**Issue by schaefi** [Azure/azure-sdk-for-python#695]
# Problem description

azure-servicemanagement-legacy has no methods to associate / disassociate a reserved ip to a deployment. REST docs: https://msdn.microsoft.com/en-us/library/azure/mt416758.aspx

# Solution

I'm going to add support for that in the SDK which will then be used in
https://github.com/SUSE/azurectl

__Current state of implementation visible here:__

* git clone https://github.com/SUSE/azure-sdk-for-python
* git checkout reserved_ip_association

**Comment by lmazuel** [comment]
Thank you :)

**Comment by schaefi** [comment]
If I only can get the recording to work :) I sent you an e-mail about my issue, hope you don't mind

**Comment by lmazuel** [comment]
@schaefi was a crazy day, sorry for the delay in response, but you were right to send me an email :). I look asap.

**Comment by lmazuel** [comment]
This PR should allow you to do your testing more easily:
https://github.com/Azure/azure-sdk-for-python/pull/699

**Comment by schaefi** [comment]
Thanks, I merged and rebased my branch so working on top of your changes. However I'm still facing an authentication problem. More details by mail. Thanks for your help

**Comment by lyan-git** [comment]
Wonder how the sdk perform the ARM public ip association and disassociation? Please let me know if you have any sample code to reference.

Thanks.
Li

**Comment by lmazuel** [comment]
For ARM, just use the NIC sdk:
```python
saved_public_ip = nic_vm1.ip_configurations[0].public_ip_address
nic_vm1.ip_configurations[0].public_ip_address = None
network_client.network_interfaces.create_or_update(rg, name1, nic_vm1) 
nic_vm2.ip_configurations[0].public_ip_address = saved_public_ip
network_client.network_interfaces.create_or_update(rg, name2, nic_vm2) 
```

And thanks @lyan-git for your tests on this ;)

This is the same approach than PowerShell:
https://azure.microsoft.com/en-us/documentation/articles/virtual-network-ip-addresses-overview-arm/#comment-2789018446

**Comment by lmazuel** [comment]
Released in 0.20.4, I authorize myself to close the issue, please tell me if there is something else to add.
Thanks again for the PR!
